department-of-veterans-affairs/vets-website
# Add max rating education to 526 rated disabilities screen

## Summary

- This PR updates the 526EZ rated disabilities screen to show text education alongside any rated disability known to already be at the maximum rating.
- I am on the Employee Experience team, under Claim Fast Tracking crew. This change was done in collaboration with the [Disability Experience team](https://github.com/orgs/department-of-veterans-affairs/teams/dbex-trex), which owns maintenance of this component.
- The new education is behind a feature flag [disability_526_maximum_rating](https://github.com/department-of-veterans-affairs/vets-api/pull/13458). The team will evaluate whether to keep the toggle on or off 30 days after launching the Max CFI rating feature.

## Related issue(s)

- Original issue: department-of-veterans-affairs/abd-vro#1858
- Adding feature flag: department-of-veterans-affairs/vets-api#13458

## Testing done

- Old behavior: no education is displayed.
- New behavior: when feature toggle is on, education is displayed alongside any rated disability if the maximum rating is available on the disability object and matches the current rating.
- Unit testing: New test added to verify that education is rendered next to one rated disability that meets the requirements, and not rendered next to another rated disability that doesn't meet them.
- Local development test: (if hitting a local vets-api, see hints below)
  - Enable the Flipper flag `disability_526_maximum_rating`
  - Sign in as ID.me user vets.gov.user+221@gmail.com at http://localhost:3001/sign-in/mocked-auth
  - Navigate to http://localhost:3001/disability/file-disability-claim-form-21-526ez/start
  - Start a new 526 flow and fill it out with sample data, selecting **I'm filling a claim for** ✅ *One or more of my rated conditions that have gotten worse* when appropriate.
  - Verify that the next screen (rated disabilities) shows the education alongside **Tinnitus** rated at 10%, and no education alongside the other rated disabilities. See screenshots 👇 
- Hints for testing against a local vets-api:
  - Ensure that vets-api-mockdata is up-to-date with department-of-veterans-affairs/vets-api-mockdata#402
  - If you already have an in-progress 526 for User 221, delete that row from the `in_progress_forms` table in Postgres before starting a new 526.
  - Mocking is imperfect. When hitting `GET /v0/disability_compensation_in_progress_forms/21-526EZ`, this [PCIU call](https://github.com/department-of-veterans-affairs/vets-api/blob/d2e887c/app/models/form_profile.rb#L348) throws an exception instead of falling back on mockdata for me, so I replaced [this line](https://github.com/department-of-veterans-affairs/vets-api/blob/d2e887c/app/models/form_profile.rb#L350) with a blanket `rescue`. Fixing PCIU mocking is way out-of-scope for this feature.

## Screenshots

### Mobile (before)
<img width="320" alt="Screen Shot 2023-08-17 at 22 25 54" src="https://github.com/department-of-veterans-affairs/vets-website/assets/282869/9eb8572f-f993-4062-93e0-e285c7a913a2">

### Mobile (after)
<img width="323" alt="Screen Shot 2023-08-17 at 22 29 14" src="https://github.com/department-of-veterans-affairs/vets-website/assets/282869/86f14cf2-c853-49ce-bcb7-c46c50331b4e">

### Desktop (before)
<img width="530" alt="Screen Shot 2023-08-17 at 22 25 11" src="https://github.com/department-of-veterans-affairs/vets-website/assets/282869/448ed40b-e011-4ff0-b915-780dfeaa0cd3">

### Desktop (after)
<img width="528" alt="Screen Shot 2023-08-17 at 22 28 58" src="https://github.com/department-of-veterans-affairs/vets-website/assets/282869/d937e625-1c6e-4bb5-a092-62d4778b1c2e">

## What areas of the site does it impact?

Form 526EZ (rated disabilities screen)

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication
